cnf-tests/docgen/cmd/fill.go

```diff
@@ -78,21 +78,49 @@ func fill(xmlFile, descriptionsFile string) {
 
 func fillDescriptions(fileName string, tests TestSuite, currentDescriptions map[string]string) error {
 	missing := false
+	removed := false
+	allTests := make([]string, 0)
 	for _, t := range tests.Testcase {
 		if _, ok := currentDescriptions[t.Name]; !ok {
 			currentDescriptions[t.Name] = emptyPlaceHolder
-			fmt.Printf("The test %s do not have a valid description\n", t.Name)
+			fmt.Printf("The test %s does not have a valid description\n", t.Name)
 			missing = true
 		}
+		allTests = append(allTests, t.Name)
+	}
+	for k := range currentDescriptions {
+		if !findAndRemoveElement(allTests, k) {
+			delete(currentDescriptions, k)
+			fmt.Printf("The test %s was removed from the test suite\n", k)
+			removed = true
+		}
 	}
 	jsonData, err := json.MarshalIndent(currentDescriptions, "", "    ")
 	err = ioutil.WriteFile(fileName, jsonData, os.ModePerm)
 	if err != nil {
 		return fmt.Errorf("failed to open the descriptions file %v", err)
 	}
+	errMissing := "Found tests with no description"
+	errRemoved := "Found tests that were removed"
+	switch {
+	case missing && removed:
+		return fmt.Errorf("%s, %s", errMissing, errRemoved)
+	case missing:
+		return fmt.Errorf(errMissing)
+	case removed:
+		return fmt.Errorf(errRemoved)
+	default:
+		return nil
+	}
+}
 
-	if missing {
-		return fmt.Errorf("Found tests with no description")
+func findAndRemoveElement(sl []string, s string) bool {
```

Comment:
please lets use a map and just go over the list one time in the insert time then we can just search with the key